### ISSUE TITLE
M2 Mac Mini / Display on HDMI does not wake up

### ISSUE THREAD

**Issue by bedawi** [AsahiLinux/docs#94]
I have installed the Fedora Asahi Remix on my M2 Mac Mini. I connected an LG 4K Display to HDMI.

Problem:
When the screen is being turned off (after locked screen or suspend) it does not turn on again. The monitor wakes up but turns off again because it does not receive a signal.

Please let me know what information you need to diagnose this issue. 
Thank you very much for bringing Linux to the Mac.

**Comment by code-mart** [comment]
I am seeing the same behavior, but I haven't been able to solve it either.

**Comment by code-mart** [comment]
Meanwhile, I have disabled suspend and automatic screen off in the settings.

**Comment by code-mart** [comment]
> Meanwhile, I have disabled suspend and automatic screen off in the settings.

Unfortunately, that apparently did nothing to solve the problem 😬

**Comment by akhanpara1** [comment]
i also have mac mini m2 when installation done after that only see asahi logo and nothing not boot ubuntu

**Comment by code-mart** [comment]
Still having this issue after updating today, but maybe this is not the correct repository for this issue to be reported in? It's not a documentation issue, after all. @marcan, should this issue be moved to a different place?

**Comment by E-man-dev** [comment]
I can report that the same thing happens with the HDMI on my 2021 MBP 16”. Works before sleep, does not send a signal after sleep.

**Comment by sithmike** [comment]
Also same issue on m2 mac mini, just installed gentoo so freshly compiled kernel, and the same thing happens

**Comment by TomasOliveiraSilva** [comment]
I installed Asahi linux yesterday on an M2 Mac mini and all went smoothly. Awesome! My Mac mini is connected via HDMI to a Dell monitor, which is also connected to another computer. I lose the HDMI signal on the Mac after telling the monitor to display the connection to the other computer; when I switch back, the display says there is no HDMI signal and enters power save mode. Likewise if I disconnect and reconnect the HDMI cable. Fortunately for me, I had already installed and activated the ssh server, so I can still work remotely. Still, this is a major nuisance. Hopefully, this will be corrected in the near future.

dmesg says (there are other messages, but this one is omnious):
[87535.364237] apple-dcp 271c00000.dcp: RTKit: syslog message: nifiedPipeline.cpp:7739: setmode failed

**Comment by cxvisa** [comment]
Same issue for my with M2 Ultra Mac Studio.  The display on HDMI does not wake up after some idle time.

**Comment by jb747** [comment]
I have been testing using a KVM with a mix of Intel and Apple Silicon Mac minis running Fedora 40. 

They all switch successfully running OSX on Apple Silicon and Fedora 40 on Intel, sometimes taking 3-4 seconds to switch.

When I boot the OSX systems to Asahi Fedora 40 server they no longer switch consistently. Sometimes the keyboard will flash. Other times it won't. Sometimes the display will switch. Other times it will announce "No signal detected". A few times, not often, the switch works. 

It seems that the Asahi implementation may abandon the switching process earlier than the other operating systems.

**Comment by jb747** [comment]
An HDMI EDID pass-through dongle which mimics the required display resolution seems to be a viable workaround. A Malarpon HDMI EDID Emulator, 3rd generation, 1920 x 1080@59 Hz plugged into the Apple Silicon Mac Mini HDMI ports enables smooth switching in my environment. There are quite a few similar devices at different price points - YMMV.

**Comment by milindnirgun** [comment]
Facing the same issue on M2 Mac Studio. Started happening after last update.

![Image](https://github.com/user-attachments/assets/708a47d4-3396-4e17-abe0-1c0e34031448)

**Comment by hamzakc** [comment]
Just installed Asahi Fedora Remix 42 on my macbook pro M2 and am facing the same issue.  Is there a command line tool we can run to reboot the hdml connection ? Apart from that I am really impressed and have moved my development over to it for a month to test it out. ❤

**Comment by milindnirgun** [comment]
I run the following command from a shell by logging in from another laptop or phone to restart plasma. And the display comes back on in a few seconds. I have noticed this problem usually happens for a long sleep session like overnight. During the day and shorter sleep duration, it wakes up normally.
Command to restart:    $ killall startplasma-wayland

**Comment by GZGavinZhao** [comment]
I appear to be having the same problem on M2 Max with kernel `6.14.8-400.asahi.fc42.aarch64+16k`. I'm using Niri and this is the log:
```
Jun 11 13:49:35 fedora niri[2092]: 2025-06-11T20:49:35.046105Z DEBUG niri::input: lid switch opened
Jun 11 13:49:35 fedora niri[2092]: 2025-06-11T20:49:35.046121Z DEBUG niri::backend::tty: connecting connector: eDP-1
Jun 11 13:49:35 fedora niri[2092]: 2025-06-11T20:49:35.046135Z DEBUG niri::backend::tty: picking mode: Mode { name: "3456x2160", clock: 521065, size: (3456, 2160), hsync: (3464,>
Jun 11 13:49:35 fedora niri[2092]: 2025-06-11T20:49:35.046144Z DEBUG niri::backend::tty: error setting max bpc: couldn't find max bpc property
Jun 11 13:49:35 fedora niri[2092]: 2025-06-11T20:49:35.046151Z DEBUG niri::backend::tty: error getting gamma properties: missing GAMMA_LUT property
Jun 11 13:49:35 fedora niri[2092]: 2025-06-11T20:49:35.046153Z DEBUG niri::backend::tty: error resetting gamma: setting gamma is not supported
Jun 11 13:49:35 fedora niri[2092]: 2025-06-11T20:49:35.048980Z DEBUG niri::niri: putting output eDP-1 at x=856 y=1440
Jun 11 13:49:35 fedora niri[2092]: 2025-06-11T20:49:35.049057Z DEBUG niri::backend::tty: device changed: 57858
Jun 11 13:49:35 fedora niri[2092]: 2025-06-11T20:49:35.049096Z DEBUG niri::backend::tty: disconnecting connector: "HDMI-A-1"
Jun 11 13:49:39 fedora niri[2092]: 2025-06-11T20:49:39.049971Z DEBUG niri::backend::tty: device changed: 57858
Jun 11 13:49:39 fedora niri[2092]: 2025-06-11T20:49:39.050037Z  WARN niri::backend::tty: error getting EDID info for HDMI-A-1: error getting EDID blob value
Jun 11 13:49:39 fedora niri[2092]: Caused by:
Jun 11 13:49:39 fedora niri[2092]:     No such file or directory (os error 2)
Jun 11 13:49:39 fedora niri[2092]: 2025-06-11T20:49:39.050043Z DEBUG niri::backend::tty: new connector: HDMI-A-1 "Unknown Unknown Unknown"
Jun 11 13:49:39 fedora niri[2092]: 2025-06-11T20:49:39.050049Z DEBUG niri::backend::tty: connecting connector: HDMI-A-1
Jun 11 13:49:39 fedora niri[2092]: 2025-06-11T20:49:39.050056Z  WARN niri::backend::tty: error connecting connector: no mode
Jun 11 13:49:41 fedora niri[2092]: 2025-06-11T20:49:41.028920Z DEBUG niri::backend::tty: device changed: 57858
Jun 11 13:49:41 fedora niri[2092]: 2025-06-11T20:49:41.029009Z DEBUG niri::backend::tty: disconnecting connector for crtc: crtc::Handle(47)
Jun 11 13:49:41 fedora niri[2092]: 2025-06-11T20:49:41.029011Z DEBUG niri::backend::tty: crtc wasn't enabled
```
I don't think the error when getting EDID info is a problem. The error message that concerns me is `error connecting connector: no mode`.

**Comment by milindnirgun** [comment]
I have switched to Cosmic desktop since I posted my last comment and not having any failure to wakeup from sleep. So it was clearly a Plasma related problem. I am happy with Cosmic for now although it is not as full featured as KDE. Other option is to go with Gnome, but I wanted to give Cosmic a try.

**Comment by fckghb** [comment]
I'm having the same issue on release 43 on a M1 Macbook Pro Max :/
Unfortunately, the trick of turning the monitor off and on does not work. Only thing that does it is a reboot.

Is there a command to restart the display interface only?

Also, has anyone tried this for USB-C output?
https://github.com/bharambetejas/asahi-fairydust-display

**Comment by milindnirgun** [comment]
Yes, you can open an ssh session from an external device, I use terminal emulator from my phone, and restart the display manager by running 'sudo systemctl restart gdm'. This should get your displays working. I try the following things in this order for best results:
1. Switch monitor off and on,
2. Restart display manager,
3. Reboot.

Most times 1 and 2 work. Rarely do I have to reboot. Hope this helps.
